### PR TITLE
Update Scanpy plot (new function)

### PR DESCRIPTION
**Pull request by heylf**
FOR CONTRIBUTOR:
* [X] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] License permits unrestricted use (educational + commercial)
* [ ] This PR adds a new tool or tool collection
* [X] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

### CODE REVIEW

**Comment by heylf** [comment]
this PR adds a new function to scanpy plot to allow a gene dot plot based for cell types based on a marker_celltype.tsv tabular file. This is useful for marker gene detection for spatial omics.

**Review by bgruening — COMMENTED** [review]
please bump the version :-)

File: tools/scanpy/plot.xml [hunk 1]
```diff
@@ -401,7 +401,21 @@ sc.pl.rank_genes_groups_violin(
 #else if $method.method == 'pl.rank_genes_groups_dotplot'
 sc.pl.rank_genes_groups_dotplot(
     @CMD_param_plot_inputs@
-    @CMD_pl_rank_genes_groups_ext@
+    #if $method.var_names.type == 'custom'
+    #set $var_names = ([x.strip() for x in str($method.var_names.var_names).split(',')])
+    var_names=$var_names,
+    #else if $method.var_names.type == 'customfile'
+    var_names={key: [v for v in list(value.values()) if pd.notna(v)] for key, value in pd.read_csv('$var_names', sep='\t', index_col=0).to_dict(orient='index').items()},
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:408]: do you again have to check the header line here?

File: tools/scanpy/plot.xml [hunk 2]
```diff
@@ -401,7 +401,21 @@ sc.pl.rank_genes_groups_violin(
...
+    #else
+    n_genes=$method.n_genes,
+    #end if
+    #if $method.groupby
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:412]: I think checking like this only works from [profile 23.0](https://docs.galaxyproject.org/en/latest/dev/schema.html#id8). The profile of this tool is 21.09. So it should be checked as ` #if $method.groupby != ''`

File: tools/scanpy/plot.xml [hunk 3]
```diff
@@ -401,7 +401,21 @@ sc.pl.rank_genes_groups_violin(
...
+    groupby='$method.groupby',
+    #end if
+    num_categories=$method.num_categories,
+    #if $method.key
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:416]: same as above

**Review by pavanvidem — COMMENTED** [review]
This PR should be merged after https://github.com/galaxyproject/tools-iuc/pull/6086

**Comment by heylf** [comment]
Added review changes

**Comment by bgruening** [comment]
Damn, we now need a rebase here.

**Pull request closed without merging**